platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java

```diff
@@ -119,17 +119,39 @@ public static boolean isExternalStorageReadable() {
   }
 
   private long nativePtr;
+  private long activeCounter;
+  private boolean wasPaused;
 
   private FileSource(String cachePath, AssetManager assetManager) {
     initialize(Mapbox.getAccessToken(), cachePath, assetManager);
   }
 
+  public void activate() {
```

Comment:
I would opt to make redundant calls possible instead of doing counting here. I suggested a patch to do so here: https://github.com/mapbox/mapbox-gl-native/pull/9750/commits/034551f1519491110b83f74662af7d4c0c509104

This will also subtly break if the file source would be paused/resumed from another place in the future.

cc @jfirebaugh

Reply:
@jfirebaugh could you add your preference on who is responsible for counting? the binding or core?

Reply:
As noted in https://github.com/mapbox/mapbox-gl-native/pull/10174#pullrequestreview-68377496, my preference is to keep `pause`/`resume` as strict as possible -- in particular callable only by the creating thread -- because they otherwise get very difficult to reason about.

Reply:
@ivovandongen can you 👍 this if you are onboard on adding this?